crash when click "add device"

Android 4.4.4
0.5.21

Here's the logcat:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nutomic.syncthingandroid/com.nutomic.syncthingandroid.activities.SettingsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.preference.Preference.setOnPreferenceChangeListener(android.preference.Preference$OnPreferenceChangeListener)' on a null object reference
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2329)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2391)
at android.app.ActivityThread.access$900(ActivityThread.java:147)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1296)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5256)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:117)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.preference.Preference.setOnPreferenceChangeListener(android.preference.Preference$OnPreferenceChangeListener)' on a null object reference
at com.nutomic.syncthingandroid.fragments.DeviceSettingsFragment.onCreate(DeviceSettingsFragment.java:77)
at android.support.v4.app.Fragment.performCreate(Fragment.java:1763)
at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:913)
at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1126)
at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:739)
at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1489)
at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:548)
at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1220)
at android.app.Activity.performStart(Activity.java:5949)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2292)
... 11 more
```

However, I don't understand this at all. The crash shouldn't occur in the first place, because the code is correct (and hasn't changed in weeks. Also, it seems I got the crash once, but then never again.

I'm seriously stumped on this one.

--- a/src/main/res/xml/device_settings_create.xml
+++ b/src/main/res/xml/device_settings_create.xml
@@ -1,27 +1,31 @@
 <?xml version="1.0" encoding="utf-8"?>
-<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
-    android:persistent="false">
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <EditTextPreference
+        android:persistent="false"
         android:key="device_id"
         android:title="@string/device_id"
         android:widgetLayout="@layout/scan_qr_code_widget" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="name"
         android:title="@string/name" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="addresses"
         android:title="@string/addresses" />
 
     <ListPreference
+        android:persistent="false"
         android:key="compression"
         android:entries="@array/compress_entries"
         android:entryValues="@array/compress_values"
         android:title="@string/compression" />
 
     <CheckBoxPreference
+        android:persistent="false"
         android:key="introducer"
         android:title="@string/introducer" />
 

--- a/src/main/res/xml/device_settings_edit.xml
+++ b/src/main/res/xml/device_settings_edit.xml
@@ -1,35 +1,41 @@
 <?xml version="1.0" encoding="utf-8"?>
-<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
-    android:persistent="false">
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <Preference
+        android:persistent="false"
         android:key="device_id"
         android:title="@string/device_id" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="name"
         android:title="@string/name" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="addresses"
         android:title="@string/addresses" />
 
     <ListPreference
+        android:persistent="false"
         android:key="compression"
         android:entries="@array/compress_entries"
         android:entryValues="@array/compress_values"
         android:title="@string/compression" />
 
     <CheckBoxPreference
+        android:persistent="false"
         android:key="introducer"
         android:title="@string/introducer" />
 
     <Preference
+        android:persistent="false"
         android:key="version"
         android:title="@string/syncthing_version_title"
         style="?android:preferenceInformationStyle" />
 
     <Preference
+        android:persistent="false"
         android:key="current_address"
         android:title="@string/current_address"
         style="?android:preferenceInformationStyle" />

--- a/src/main/res/xml/folder_settings_create.xml
+++ b/src/main/res/xml/folder_settings_create.xml
@@ -1,33 +1,39 @@
 <?xml version="1.0" encoding="utf-8"?>
-<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
-    android:persistent="false">
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <EditTextPreference
+        android:persistent="false"
         android:key="folder_id"
         android:title="@string/folder_id" />
 
     <Preference
+        android:persistent="false"
         android:key="directory"
         android:title="@string/directory" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="rescan_interval"
         android:title="@string/rescan_interval"
         android:inputType="numberDecimal" />
 
     <CheckBoxPreference
+        android:persistent="false"
         android:key="folder_master"
         android:title="@string/folder_master" />
 
     <PreferenceScreen
+        android:persistent="false"
         android:key="devices"
         android:title="@string/devices" />
 
     <CheckBoxPreference
+        android:persistent="false"
         android:key="versioning"
         android:title="@string/file_versioning" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="versioning_keep"
         android:title="@string/keep_versions"
         android:inputType="numberDecimal" />

--- a/src/main/res/xml/folder_settings_edit.xml
+++ b/src/main/res/xml/folder_settings_edit.xml
@@ -1,37 +1,43 @@
 <?xml version="1.0" encoding="utf-8"?>
-<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
-    android:persistent="false">
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <EditTextPreference
+        android:persistent="false"
         android:key="folder_id"
         android:title="@string/folder_id"
         android:enabled="false"
         style="?android:preferenceInformationStyle" />
 
     <Preference
+        android:persistent="false"
         android:key="directory"
         android:title="@string/directory"
         android:enabled="false"
         style="?android:preferenceInformationStyle" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="rescan_interval"
         android:title="@string/rescan_interval"
         android:inputType="numberDecimal" />
 
     <CheckBoxPreference
+        android:persistent="false"
         android:key="folder_master"
         android:title="@string/folder_master" />
 
     <PreferenceScreen
+        android:persistent="false"
         android:key="devices"
         android:title="@string/devices" />
 
     <CheckBoxPreference
+        android:persistent="false"
         android:key="versioning"
         android:title="File Versioning" />
 
     <EditTextPreference
+        android:persistent="false"
         android:key="versioning_keep"
         android:title="@string/keep_versions"
         android:inputType="numberDecimal" />

